pyflagser/flagser.py

```diff
@@ -1,40 +1,150 @@
 """Implementation of the python API for the flagser C++ library."""
 
 import numpy as np
+
+from ._utils import _extract_unweighted_graph, _extract_weighted_graph
 from flagser_pybind import compute_homology, implemented_filtrations
 
 
-def flagser(flag_matrix, min_dimension=0, max_dimension=np.inf, directed=True,
-            filtration="max", coeff=2, approximation=-1):
+def flagser_unweighted(adjacency_matrix, min_dimension=0, max_dimension=np.inf,
+                       directed=True, coeff=2, approximation=None):
+    """Compute homology of a directed/undirected unweighted flag complex.
+
+    From an `adjacency_matrix` construct all cells forming its associated flag
+    complex and compute its homology.
+
+    Parameters
+    ----------
+    adjacency_matrix : 2d ndarray or scipy.sparse matrix, required
+        Adjacency matrix of a directed/undirected unweighted graph. It is
+        understood as a boolean matrix. Off-diagonal, ``0`` or ``False`` values
+        denote abstent edges while non-``0`` or ``True`` values denote edges
+        which are present. Diagonal values are ignored.
+
+    min_dimension : int, optional, default: ``0``
+        Minimum homology dimension to compute.
+
+    max_dimension : int or np.inf, optional, default: ``np.inf``
+        Maximum homology dimension to compute.
+
+    directed : bool, optional, default: ``True``
+        If ``True``, computes homology for the directed flad complex determined
+        by `adjacency_matrix`. If ``False``, computes homology for the
+        undirected flag complex obtained by considering all edges as
+        undirected.
+
+    coeff : int, optional, default: ``2``
+        Compute homology with coefficients in the prime field
+        :math:`\\mathbb{F}_p = \\{ 0, \\ldots, p - 1 \\}` where
+        :math:`p` equals `coeff`.
+
+    approximation : int or None, optional, default: ``None``
+        Skip all cells creating columns in the reduction matrix with more than
+        this number of entries. Use this for hard problems; a good value is
+        often ``100,000``. Increase for higher precision, decrease for faster
+        computation. If ``None``, no approximation is made and all cells are
+        used. For more details, please refer to [1]_.
+
+    Returns
+    -------
+    out : dict of list
+        A dictionary with the following key-value pairs:
+
+        - ``'cell_count'``: list of int
```

Comment:
Would the typical user care a lot about ``'cell_count'`` relative to the other keywords? I know this is a dictionary and order does not matter, but still: if we expect this key to be less used, how about placing it later in the doctrsing? E.g. first betti, then euler, then cell_count.

Reply:
Most people I have gotten feedback from are interested in pyflagser mostly for `cell_count`.

I would say `betti` first, then `cell_count`, then `euler`.

Reply:
Interesting, I should read what it means then :) happy with the suggested reordering.